Fresh project with barista dependency does not compile

# Bug Report

<!---
Hi there!

You are in the process of reporting a bug in Barista. To ensure a quick
turnaround, please enter/change/update the following fields:
-->

## Expected Behavior

<!--- Tell us what should happen -->
I am able to re-create one of the examples shown on https://barista.dynatrace.com/components/filter-field locally without any compilation errors.

## Current Behavior

<!--- Tell us what happens instead of the expected behavior -->
When creating a fresh local project and adding the dependency to barista using `ng add @dynatrace/barista-components` and trying to use e.g. the filter field, Angular will not compile the project without errors.

## Steps to Reproduce

<!--- Provide a link to a live example, or an unambiguous set of steps to -->
<!--- reproduce this bug. Include code to reproduce, if relevant -->

1. Create a new angular project using `ng new` and using the default options.
2. Add Barista Components to the project by using `ng add @dynatrace/barista-components`
3. Modify the auto-generated **app.module.ts** and add `DtFilterFieldModule` to the `imports` array.

## Context (Environment)

<!--- How has this issue affected you? What are you trying to accomplish? -->
<!--- Providing context helps us come up with a solution that is most useful in the real world -->

### Used Versions:

- **node**: 14.9.0
- **npm**: 6.14.8
- **typescript**: ~4.0.2
- **angular**: 10.1.0
- **@angular/cdk**: 10.1.1
- **@dynatrace/barista-components**: 8.0.0

## Attachments

<!-- Not obligatory, Append a screenshot, and any other visuals and required files if possible. -->

#### The error displayed when trying to run `ng serve`
```
    ERROR in node_modules/@dynatrace/barista-components/button-group/src/button-group.d.ts:35:9 - error TS2611: 'disabled' is defined as a property in class 'HasTabIndex & DtButtonGroupBase', but is overridden here in 'DtButtonGroup<T>' as an accessor.

    35     get disabled(): boolean;
               ~~~~~~~~
    node_modules/@dynatrace/barista-components/button-group/src/button-group.d.ts:76:9 - error TS2611: 'disabled' is defined as a property in class 'HasTabIndex & CanColor<DtButtonGroupThemePalette> & DtButtonGroupItemBase', but is overridden here in 'DtButtonGroupItem<T>' as an accessor.

    76     get disabled(): boolean;
               ~~~~~~~~
```

<!--
We at Dynatrace have a bugs first policy. Meaning, we will handle the bug as
soon as possible, depending on the severity of the bug.

Thank you and all the best! — The Dynatrace DesignOps team
-->

--- a/libs/barista-components/button-group/src/button-group.ts
+++ b/libs/barista-components/button-group/src/button-group.ts
@@ -41,12 +41,12 @@ import {
   mixinColor,
   mixinTabIndex,
   _readKeyCode,
+  HasElementRef,
 } from '@dynatrace/barista-components/core';
 
-export class DtButtonGroupBase {
-  disabled: boolean;
-}
-export const _DtButtonGroup = mixinTabIndex(DtButtonGroupBase);
+export const _DtButtonGroup = mixinTabIndex(
+  class {} as Constructor<CanDisable>,
+);
 
 @Component({
   selector: 'dt-button-group',
@@ -156,15 +156,14 @@ export interface DtButtonGroupItemSelectionChange<T> {
 
 export type DtButtonGroupThemePalette = 'main' | 'error';
 export class DtButtonGroupItemBase {
-  /** Whether the button group item is disabled. */
-  disabled: boolean;
   constructor(public _elementRef: ElementRef) {}
 }
+
 export const _DtButtonGroupItem = mixinTabIndex(
-  mixinColor<Constructor<DtButtonGroupItemBase>, DtButtonGroupThemePalette>(
-    DtButtonGroupItemBase,
-    'main',
-  ),
+  mixinColor<
+    Constructor<CanDisable & HasElementRef>,
+    DtButtonGroupThemePalette
+  >(DtButtonGroupItemBase as Constructor<CanDisable & HasElementRef>, 'main'),
 );
 
 @Component({
@@ -247,7 +246,8 @@ export class DtButtonGroupItem<T> extends _DtButtonGroupItem
   constructor(
     private _buttonGroup: DtButtonGroup<T>,
     private _changeDetectorRef: ChangeDetectorRef,
-    _elementRef: ElementRef,
+    /** @internal */
+    public _elementRef: ElementRef,
     private _focusMonitor: FocusMonitor,
   ) {
     super(_elementRef);

--- a/libs/barista-components/core/src/common-behaviours/color.ts
+++ b/libs/barista-components/core/src/common-behaviours/color.ts
@@ -70,7 +70,6 @@ export function mixinColor<
     // tslint:disable-next-line:no-any
     constructor(...args: any[]) {
       super(...args);
-
       // Set the default color that can be specified from the mixin.
       this.color = defaultColor as P;
     }

